src/undo/PageLayerPosEntry.h

```diff
@@ -11,22 +11,17 @@
 
 #pragma once
 
-template <class T>
-class PageLayerPosEntry {
-public:
-    PageLayerPosEntry(Layer* layer, T* element, int pos) {
-        this->element = element;
-        this->pos = pos;
-        this->layer = layer;
-    }
+class Layer;
 
-    ~PageLayerPosEntry() {}
+template <class T>
+struct PageLayerPosEntry {
+    // TODO: constructor could be removed with C++20
+    PageLayerPosEntry<T>(Layer* layer, T* element, int pos): layer(layer), element(element), pos(pos) {}
```

Comment:
Are you sure? I think this is already replaceable since c++14 or at least c++17.

Reply:
At least `vector::emplace` does not support it yet.
But I think this is cleaner than dealing with it in the files using it.